chain/chain/src/chain.rs

```diff
@@ -2440,8 +2440,8 @@ impl<'a> ChainUpdate<'a> {
         F: FnMut(ChallengeBody) -> (),
     {
         // Refuse blocks from the too distant future.
-        if header.timestamp() > Utc::now() + Duration::seconds(ACCEPTABLE_TIME_DIFFERENCE) {
-            return Err(ErrorKind::InvalidBlockFutureTime(header.timestamp()).into());
+        if header.timestamp() > Utc::now() + Duration::seconds(ACCEPTABLE_FUTURE_TIME_DIFFERENCE) {
+            return Err(ErrorKind::Orphan.into());
```

Comment:
At which point will the block get unorphaned?

Reply:
In `check_orphans` we run `process_single_block` again and it at that point the time is valid the block is unorphaned

Reply:
`check_orphans` only processes blocks that build on top of `prev_hash`. Naturally, given the previous block is likely known at the time you orphans the current block, we will never call `check_orphans` with the `prev_hash` of the current block.